refactor(deployments): convert environment object to a string

c-koehler opened the pull request:
fixes https://github.com/openshiftio/openshift.io/issues/2585

c-koehler commented:
NOTE: Should not be merged until after I post another message, as I have to test something when I get in Monday morning and get OSIO working properly.

andrewazores commented on src/app/space/create/deployments/services/deployments.service.ts:
```diff
@@ -240,18 +239,17 @@ export class DeploymentsService implements OnDestroy {
       )
-      .distinctUntilChanged((p: ModelEnvironment[], q: ModelEnvironment[]) =>
-        deepEqual(new Set<string>(p.map(v => v.name)), new Set<string>(q.map(v => v.name))));
+      .distinctUntilChanged((p: string[], q: string[]) => deepEqual(new Set<string>(p), new Set<string>(q)));
```
Can you test if this custom equality/comparison function used for `distinctUntilChanged` is still needed? The original reason it was included rather than using the default was because arrays of Environment objects did not compare as equal as expected (ie. [{ name: 'foo' }] != [{ name: 'foo' }]). Now that they're just string arrays I suspect it'll work as expected by just doing `.distinctUntilChanged()` with no comparison function argument.

andrewazores commented:
Testing this, the resource usage panel is broken. There's at least still one place where a correction needs to be made: resource-card.component.html .

andrewazores commented:
Also please update the commit message to include the full link to the openshiftio/openshift.io issue (as I have done to the initial comment of this PR) rather than simply the number reference. Since that number reference is to the an issue tracker in a different repository the reference becomes invalid when used here. #2586 is an unrelated PR when used here, what's really meant is `https://github.com/openshiftio/openshift.io/issues/2586`

c-koehler commented:
@andrewazores Updated the following:
- `distinctUntilChanged()` appears to work without the custom comparison
- fixed the openshift.io link to point to the proper issue (nice catch)
- fixed the HTML files (it happened to span a few), running it will show it functioning now

c-koehler commented:
I should also have said it's also not a WIP anymore, this should be operational.

fabric8cd commented:
@c-koehler fabric8/fabric8-ui:SNAPSHOT-PR-2644-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2644-fabric8-ui.badger.fabric8.io

andrewazores commented:
Note: merge https://github.com/fabric8-ui/fabric8-ui/pull/2643 before this one, rebasing in that direction should be easier

joshuawilson commented:
@c-koehler please rebase

c-koehler commented:
@joshuawilson rebased and ready to go!

fabric8cd commented:
@c-koehler fabric8/fabric8-ui:SNAPSHOT-PR-2644-5 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2644-fabric8-ui.badger.fabric8.io

joshuawilson commented:
@c-koehler looks like it needs another rebase

c-koehler commented:
@joshuawilson rebased!

fabric8cd commented:
@c-koehler fabric8/fabric8-ui:SNAPSHOT-PR-2644-6 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2644-fabric8-ui.badger.fabric8.io
